target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-mf289f.dts

```diff
@@ -0,0 +1,436 @@
+                };
+        };
+
+        keys {
+                compatible = "gpio-keys";
+
+                reset {
+                        label = "reset";
+                        linux,code = <KEY_RESTART>;
+                        gpios = <&tlmm 18 GPIO_ACTIVE_LOW>;
+                };
+
+                wps {
+                        label = "wps";
+                        linux,code = <KEY_WPS_BUTTON>;
+                        gpios = <&tlmm 68 GPIO_ACTIVE_LOW>;
+                };
+        };
+
+        soc {
+                rng@22000 {
+                        status = "okay";
+                };
+
+                mdio@90000 {
+                        status = "okay";
+                        pinctrl-0 = <&mdio_pins>;
+                        pinctrl-names = "default";
+                        reset-gpios = <&tlmm 47 GPIO_ACTIVE_LOW>;
+                        reset-delay-us = <2000>;
+                };
+
+                ess-psgmii@98000 {
+                        status = "okay";
+                };
+
+                tcsr@1949000 {
+                        compatible = "qcom,tcsr";
+                        reg = <0x1949000 0x100>;
+                        qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+                };
+
+                tcsr@194b000 {
+                        /* select hostmode */
+                        compatible = "qcom,tcsr";
+                        reg = <0x194b000 0x100>;
+                        qcom,usb-hsphy-mode-select = <TCSR_USB_HSPHY_HOST_MODE>;
+                        status = "okay";
+                };
+
+                ess_tcsr@1953000 {
+                        compatible = "qcom,tcsr";
+                        reg = <0x1953000 0x1000>;
+                        qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+                };
+
+                tcsr@1957000 {
+                        compatible = "qcom,tcsr";
+                        reg = <0x1957000 0x100>;
+                        qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+                };
+
+                usb2@60f8800 {
+                        status = "okay";
+                };
+
+                usb3@8af8800 {
+                        status = "okay";
+                };
+
+                crypto@8e3a000 {
+                        status = "okay";
+                };
+
+                watchdog@b017000 {
+                        status = "okay";
+                };
+
+                ess-switch@c000000 {
+                        status = "okay";
+                };
+
+                edma@c080000 {
+                        status = "okay";
+                };
+        };
+};
+
+&blsp_dma {
+        status = "okay";
+};
+
+&blsp1_spi1 {
+        pinctrl-0 = <&spi_0_pins>;
+        pinctrl-names = "default";
+        status = "okay";
+        cs-gpios = <&tlmm 12 GPIO_ACTIVE_HIGH>,
+                   <&tlmm 54 GPIO_ACTIVE_HIGH>;
+
+        flash@0 {
+                compatible = "jedec,spi-nor", "n25q128a11";

```

Comment:
What do you need "n25q128a11" for?
It's QCA-ism that needs to be removed from existence.

Reply:
@robimarko IIRC U-boot was looking for that string, at least on MF286D, which is predecessor of this device. Without that device would refuse to boot. Please look at MF286D's device tree.

@stich86 could you check if this can be removed?

Reply:
looks like it works also without fix the NOR type

```
[    1.182132] spi_qup 78b5000.spi: IN:block:16, fifo:64, OUT:block:16, fifo:64
[    1.184763] spi-nor spi0.0: gd25q16 (2048 Kbytes)
[    1.190461] 8 fixed-partitions partitions found on MTD device spi0.0
[    1.194953] Creating 8 MTD partitions on "spi0.0":
[    1.201394] 0x000000000000-0x000000040000 : "0:SBL1"
[    1.206542] 0x000000040000-0x000000060000 : "0:MIBIB"
[    1.211608] 0x000000060000-0x0000000c0000 : "0:QSEE"
[    1.216514] 0x0000000c0000-0x0000000d0000 : "0:CDT"
[    1.221601] 0x0000000d0000-0x0000000e0000 : "0:DDRPARAMS"
[    1.226267] 0x0000000e0000-0x0000000f0000 : "0:APPSBLENV"
[    1.231807] 0x0000000f0000-0x0000001b0000 : "0:APPSBL"
[    1.237092] 0x0000001b0000-0x000000200000 : "0:reserved1"
```
i'm removing from DTS

Reply:
If it works fine, remove it.

But yeah, if there are any quirks needed just add a comment for it, otherwise I have no idea that its really required.